codegen/smithy-aws-typescript-codegen/src/main/java/software/amazon/smithy/aws/typescript/codegen/XmlShapeDeserVisitor.java

```diff
@@ -164,16 +164,17 @@ void deserializeNamedMember(
 
         // Handle self-closed xml parsed as an empty string.
         if (deserializationReturnsArray) {
-            writer.openBlock("if ($L.$L === \"\") {", "}", inputLocation, locationName, () -> {
+            writer.openBlock("if ($L['$L'] === \"\") {", "}", inputLocation, locationName, () -> {
```

Comment:
`'$L'` should just be `$S` if this change is necessary.

Reply:
Not needed.  Reverted in 593308c84.